[2024-09-09T16:08:58.656-0300] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-09-09T16:08:58.669-0300] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: dag_save_database_raw.dag_save_database_raw scheduled__2024-08-21T08:15:00+00:00 [queued]>
[2024-09-09T16:08:58.678-0300] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: dag_save_database_raw.dag_save_database_raw scheduled__2024-08-21T08:15:00+00:00 [queued]>
[2024-09-09T16:08:58.679-0300] {taskinstance.py:2865} INFO - Starting attempt 1 of 11
[2024-09-09T16:08:58.707-0300] {taskinstance.py:2888} INFO - Executing <Task(BashOperator): dag_save_database_raw> on 2024-08-21 08:15:00+00:00
[2024-09-09T16:08:58.710-0300] {standard_task_runner.py:72} INFO - Started process 132727 to run task
[2024-09-09T16:08:58.711-0300] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'dag_save_database_raw', 'dag_save_database_raw', 'scheduled__2024-08-21T08:15:00+00:00', '--job-id', '31', '--raw', '--subdir', 'DAGS_FOLDER/dag_save_database_raw.py', '--cfg-path', '/tmp/tmp0f33_tbd']
[2024-09-09T16:08:58.713-0300] {standard_task_runner.py:105} INFO - Job 31: Subtask dag_save_database_raw
[2024-09-09T16:08:58.771-0300] {task_command.py:467} INFO - Running <TaskInstance: dag_save_database_raw.dag_save_database_raw scheduled__2024-08-21T08:15:00+00:00 [running]> on host DESKTOP-0CT35O1.
[2024-09-09T16:08:58.853-0300] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Matheus' AIRFLOW_CTX_DAG_ID='dag_save_database_raw' AIRFLOW_CTX_TASK_ID='dag_save_database_raw' AIRFLOW_CTX_EXECUTION_DATE='2024-08-21T08:15:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-08-21T08:15:00+00:00'
[2024-09-09T16:08:58.855-0300] {logging_mixin.py:190} INFO - Task instance is in running state
[2024-09-09T16:08:58.856-0300] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2024-09-09T16:08:58.858-0300] {logging_mixin.py:190} INFO - Current task name:dag_save_database_raw state:running start_date:2024-09-09 19:08:58.670930+00:00
[2024-09-09T16:08:58.859-0300] {logging_mixin.py:190} INFO - Dag name:dag_save_database_raw and current dag run status:running
[2024-09-09T16:08:58.860-0300] {taskinstance.py:731} INFO - ::endgroup::
[2024-09-09T16:08:58.862-0300] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2024-09-09T16:08:58.865-0300] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', "python '/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/save_database.py'"]
[2024-09-09T16:08:58.869-0300] {subprocess.py:86} INFO - Output:
[2024-09-09T16:09:03.574-0300] {subprocess.py:93} INFO - 24/09/09 16:09:03 WARN Utils: Your hostname, DESKTOP-0CT35O1 resolves to a loopback address: 127.0.1.1; using 10.255.255.254 instead (on interface lo)
[2024-09-09T16:09:03.577-0300] {subprocess.py:93} INFO - 24/09/09 16:09:03 WARN Utils: Set SPARK_LOCAL_IP if you need to bind to another address
[2024-09-09T16:09:04.066-0300] {subprocess.py:93} INFO - 24/09/09 16:09:04 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-09-09T16:09:04.227-0300] {subprocess.py:93} INFO - Setting default log level to "WARN".
[2024-09-09T16:09:04.228-0300] {subprocess.py:93} INFO - To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
[2024-09-09T16:09:11.346-0300] {subprocess.py:93} INFO - [Stage 1:>                                                          (0 + 1) / 1]                                                                                24/09/09 16:09:11 WARN SparkStringUtils: Truncated the string representation of a plan since it was too large. This behavior can be adjusted by setting 'spark.sql.debug.maxToStringFields'.
[2024-09-09T16:09:11.709-0300] {subprocess.py:93} INFO - 24/09/09 16:09:11 ERROR Executor: Exception in task 0.0 in stage 2.0 (TID 2)
[2024-09-09T16:09:11.710-0300] {subprocess.py:93} INFO - org.apache.spark.SparkException: Parquet column cannot be converted in file file:///mnt/c/users/teu20/Documents/Poli/Dados/Raw/empregados/part-00000-ef92445d-d373-4a30-bbf7-dd1f523fde46-c000.snappy.parquet. Column: [reviews_count], Expected: int, Found: BINARY.
[2024-09-09T16:09:11.712-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.errors.QueryExecutionErrors$.unsupportedSchemaColumnConvertError(QueryExecutionErrors.scala:855)
[2024-09-09T16:09:11.713-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:287)
[2024-09-09T16:09:11.714-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-09-09T16:09:11.715-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.FileSourceScanExec$$anon$1.hasNext(DataSourceScanExec.scala:593)
[2024-09-09T16:09:11.716-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.columnartorow_nextBatch_0$(Unknown Source)
[2024-09-09T16:09:11.718-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-09-09T16:09:11.721-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-09-09T16:09:11.723-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-09-09T16:09:11.724-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-09-09T16:09:11.725-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-09-09T16:09:11.727-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator.isEmpty(Iterator.scala:387)
[2024-09-09T16:09:11.728-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator.isEmpty$(Iterator.scala:387)
[2024-09-09T16:09:11.730-0300] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.isEmpty(Iterator.scala:1431)
[2024-09-09T16:09:11.731-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.savePartition(JdbcUtils.scala:684)
[2024-09-09T16:09:11.733-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.$anonfun$saveTable$1(JdbcUtils.scala:904)
[2024-09-09T16:09:11.734-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.$anonfun$saveTable$1$adapted(JdbcUtils.scala:903)
[2024-09-09T16:09:11.735-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2(RDD.scala:1039)
[2024-09-09T16:09:11.736-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2$adapted(RDD.scala:1039)
[2024-09-09T16:09:11.737-0300] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.$anonfun$runJob$5(SparkContext.scala:2433)
[2024-09-09T16:09:11.739-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-09-09T16:09:11.740-0300] {subprocess.py:93} INFO - 	at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-09-09T16:09:11.741-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-09-09T16:09:11.743-0300] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-09-09T16:09:11.744-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-09-09T16:09:11.745-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-09-09T16:09:11.746-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-09-09T16:09:11.747-0300] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-09-09T16:09:11.748-0300] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2024-09-09T16:09:11.749-0300] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2024-09-09T16:09:11.750-0300] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2024-09-09T16:09:11.751-0300] {subprocess.py:93} INFO - Caused by: org.apache.spark.sql.execution.datasources.SchemaColumnConvertNotSupportedException: column: [reviews_count], physicalType: BINARY, logicalType: int
[2024-09-09T16:09:11.752-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.ParquetVectorUpdaterFactory.constructConvertNotSupportedException(ParquetVectorUpdaterFactory.java:1136)
[2024-09-09T16:09:11.753-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.ParquetVectorUpdaterFactory.getUpdater(ParquetVectorUpdaterFactory.java:199)
[2024-09-09T16:09:11.754-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedColumnReader.readBatch(VectorizedColumnReader.java:175)
[2024-09-09T16:09:11.755-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader.nextBatch(VectorizedParquetRecordReader.java:342)
[2024-09-09T16:09:11.756-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader.nextKeyValue(VectorizedParquetRecordReader.java:233)
[2024-09-09T16:09:11.757-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.RecordReaderIterator.hasNext(RecordReaderIterator.scala:39)
[2024-09-09T16:09:11.758-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-09-09T16:09:11.759-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:283)
[2024-09-09T16:09:11.760-0300] {subprocess.py:93} INFO - 	... 28 more
[2024-09-09T16:09:11.761-0300] {subprocess.py:93} INFO - 24/09/09 16:09:11 WARN TaskSetManager: Lost task 0.0 in stage 2.0 (TID 2) (10.255.255.254 executor driver): org.apache.spark.SparkException: Parquet column cannot be converted in file file:///mnt/c/users/teu20/Documents/Poli/Dados/Raw/empregados/part-00000-ef92445d-d373-4a30-bbf7-dd1f523fde46-c000.snappy.parquet. Column: [reviews_count], Expected: int, Found: BINARY.
[2024-09-09T16:09:11.762-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.errors.QueryExecutionErrors$.unsupportedSchemaColumnConvertError(QueryExecutionErrors.scala:855)
[2024-09-09T16:09:11.763-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:287)
[2024-09-09T16:09:11.764-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-09-09T16:09:11.765-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.FileSourceScanExec$$anon$1.hasNext(DataSourceScanExec.scala:593)
[2024-09-09T16:09:11.766-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.columnartorow_nextBatch_0$(Unknown Source)
[2024-09-09T16:09:11.767-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-09-09T16:09:11.768-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-09-09T16:09:11.769-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-09-09T16:09:11.770-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-09-09T16:09:11.771-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-09-09T16:09:11.772-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator.isEmpty(Iterator.scala:387)
[2024-09-09T16:09:11.773-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator.isEmpty$(Iterator.scala:387)
[2024-09-09T16:09:11.774-0300] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.isEmpty(Iterator.scala:1431)
[2024-09-09T16:09:11.775-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.savePartition(JdbcUtils.scala:684)
[2024-09-09T16:09:11.776-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.$anonfun$saveTable$1(JdbcUtils.scala:904)
[2024-09-09T16:09:11.777-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.$anonfun$saveTable$1$adapted(JdbcUtils.scala:903)
[2024-09-09T16:09:11.778-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2(RDD.scala:1039)
[2024-09-09T16:09:11.779-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2$adapted(RDD.scala:1039)
[2024-09-09T16:09:11.780-0300] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.$anonfun$runJob$5(SparkContext.scala:2433)
[2024-09-09T16:09:11.781-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-09-09T16:09:11.783-0300] {subprocess.py:93} INFO - 	at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-09-09T16:09:11.784-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-09-09T16:09:11.785-0300] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-09-09T16:09:11.785-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-09-09T16:09:11.786-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-09-09T16:09:11.787-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-09-09T16:09:11.789-0300] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-09-09T16:09:11.790-0300] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2024-09-09T16:09:11.791-0300] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2024-09-09T16:09:11.791-0300] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2024-09-09T16:09:11.792-0300] {subprocess.py:93} INFO - Caused by: org.apache.spark.sql.execution.datasources.SchemaColumnConvertNotSupportedException: column: [reviews_count], physicalType: BINARY, logicalType: int
[2024-09-09T16:09:11.793-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.ParquetVectorUpdaterFactory.constructConvertNotSupportedException(ParquetVectorUpdaterFactory.java:1136)
[2024-09-09T16:09:11.795-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.ParquetVectorUpdaterFactory.getUpdater(ParquetVectorUpdaterFactory.java:199)
[2024-09-09T16:09:11.796-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedColumnReader.readBatch(VectorizedColumnReader.java:175)
[2024-09-09T16:09:11.797-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader.nextBatch(VectorizedParquetRecordReader.java:342)
[2024-09-09T16:09:11.798-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader.nextKeyValue(VectorizedParquetRecordReader.java:233)
[2024-09-09T16:09:11.798-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.RecordReaderIterator.hasNext(RecordReaderIterator.scala:39)
[2024-09-09T16:09:11.800-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-09-09T16:09:11.801-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:283)
[2024-09-09T16:09:11.802-0300] {subprocess.py:93} INFO - 	... 28 more
[2024-09-09T16:09:11.803-0300] {subprocess.py:93} INFO - 
[2024-09-09T16:09:11.804-0300] {subprocess.py:93} INFO - 24/09/09 16:09:11 ERROR TaskSetManager: Task 0 in stage 2.0 failed 1 times; aborting job
[2024-09-09T16:09:11.835-0300] {subprocess.py:93} INFO - Traceback (most recent call last):
[2024-09-09T16:09:11.836-0300] {subprocess.py:93} INFO -   File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/save_database.py", line 65, in <module>
[2024-09-09T16:09:11.838-0300] {subprocess.py:93} INFO -     dfempregados.write.jdbc(url, "empregados", mode="overwrite", properties=myproperties)
[2024-09-09T16:09:11.839-0300] {subprocess.py:93} INFO -   File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/pyspark/sql/readwriter.py", line 1984, in jdbc
[2024-09-09T16:09:11.840-0300] {subprocess.py:93} INFO -     self.mode(mode)._jwrite.jdbc(url, table, jprop)
[2024-09-09T16:09:11.841-0300] {subprocess.py:93} INFO -   File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/py4j/java_gateway.py", line 1322, in __call__
[2024-09-09T16:09:11.842-0300] {subprocess.py:93} INFO -     return_value = get_return_value(
[2024-09-09T16:09:11.843-0300] {subprocess.py:93} INFO -   File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/pyspark/errors/exceptions/captured.py", line 179, in deco
[2024-09-09T16:09:11.844-0300] {subprocess.py:93} INFO -     return f(*a, **kw)
[2024-09-09T16:09:11.845-0300] {subprocess.py:93} INFO -   File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/py4j/protocol.py", line 326, in get_return_value
[2024-09-09T16:09:11.846-0300] {subprocess.py:93} INFO -     raise Py4JJavaError(
[2024-09-09T16:09:11.903-0300] {subprocess.py:93} INFO - py4j.protocol.Py4JJavaError: An error occurred while calling o55.jdbc.
[2024-09-09T16:09:11.905-0300] {subprocess.py:93} INFO - : org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 2.0 failed 1 times, most recent failure: Lost task 0.0 in stage 2.0 (TID 2) (10.255.255.254 executor driver): org.apache.spark.SparkException: Parquet column cannot be converted in file file:///mnt/c/users/teu20/Documents/Poli/Dados/Raw/empregados/part-00000-ef92445d-d373-4a30-bbf7-dd1f523fde46-c000.snappy.parquet. Column: [reviews_count], Expected: int, Found: BINARY.
[2024-09-09T16:09:11.906-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.errors.QueryExecutionErrors$.unsupportedSchemaColumnConvertError(QueryExecutionErrors.scala:855)
[2024-09-09T16:09:11.907-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:287)
[2024-09-09T16:09:11.908-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-09-09T16:09:11.909-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.FileSourceScanExec$$anon$1.hasNext(DataSourceScanExec.scala:593)
[2024-09-09T16:09:11.910-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.columnartorow_nextBatch_0$(Unknown Source)
[2024-09-09T16:09:11.911-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-09-09T16:09:11.912-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-09-09T16:09:11.913-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-09-09T16:09:11.914-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-09-09T16:09:11.915-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-09-09T16:09:11.916-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator.isEmpty(Iterator.scala:387)
[2024-09-09T16:09:11.917-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator.isEmpty$(Iterator.scala:387)
[2024-09-09T16:09:11.919-0300] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.isEmpty(Iterator.scala:1431)
[2024-09-09T16:09:11.920-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.savePartition(JdbcUtils.scala:684)
[2024-09-09T16:09:11.920-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.$anonfun$saveTable$1(JdbcUtils.scala:904)
[2024-09-09T16:09:11.922-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.$anonfun$saveTable$1$adapted(JdbcUtils.scala:903)
[2024-09-09T16:09:11.923-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2(RDD.scala:1039)
[2024-09-09T16:09:11.924-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2$adapted(RDD.scala:1039)
[2024-09-09T16:09:11.925-0300] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.$anonfun$runJob$5(SparkContext.scala:2433)
[2024-09-09T16:09:11.926-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-09-09T16:09:11.927-0300] {subprocess.py:93} INFO - 	at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-09-09T16:09:11.928-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-09-09T16:09:11.929-0300] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-09-09T16:09:11.931-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-09-09T16:09:11.932-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-09-09T16:09:11.933-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-09-09T16:09:11.934-0300] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-09-09T16:09:11.935-0300] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2024-09-09T16:09:11.936-0300] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2024-09-09T16:09:11.937-0300] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2024-09-09T16:09:11.938-0300] {subprocess.py:93} INFO - Caused by: org.apache.spark.sql.execution.datasources.SchemaColumnConvertNotSupportedException: column: [reviews_count], physicalType: BINARY, logicalType: int
[2024-09-09T16:09:11.939-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.ParquetVectorUpdaterFactory.constructConvertNotSupportedException(ParquetVectorUpdaterFactory.java:1136)
[2024-09-09T16:09:11.941-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.ParquetVectorUpdaterFactory.getUpdater(ParquetVectorUpdaterFactory.java:199)
[2024-09-09T16:09:11.942-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedColumnReader.readBatch(VectorizedColumnReader.java:175)
[2024-09-09T16:09:11.943-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader.nextBatch(VectorizedParquetRecordReader.java:342)
[2024-09-09T16:09:11.944-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader.nextKeyValue(VectorizedParquetRecordReader.java:233)
[2024-09-09T16:09:11.946-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.RecordReaderIterator.hasNext(RecordReaderIterator.scala:39)
[2024-09-09T16:09:11.947-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-09-09T16:09:11.948-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:283)
[2024-09-09T16:09:11.949-0300] {subprocess.py:93} INFO - 	... 28 more
[2024-09-09T16:09:11.950-0300] {subprocess.py:93} INFO - 
[2024-09-09T16:09:11.951-0300] {subprocess.py:93} INFO - Driver stacktrace:
[2024-09-09T16:09:11.952-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2856)
[2024-09-09T16:09:11.953-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2792)
[2024-09-09T16:09:11.954-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2791)
[2024-09-09T16:09:11.955-0300] {subprocess.py:93} INFO - 	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2024-09-09T16:09:11.956-0300] {subprocess.py:93} INFO - 	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2024-09-09T16:09:11.958-0300] {subprocess.py:93} INFO - 	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2024-09-09T16:09:11.959-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2791)
[2024-09-09T16:09:11.960-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1247)
[2024-09-09T16:09:11.961-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1247)
[2024-09-09T16:09:11.962-0300] {subprocess.py:93} INFO - 	at scala.Option.foreach(Option.scala:407)
[2024-09-09T16:09:11.963-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1247)
[2024-09-09T16:09:11.964-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:3060)
[2024-09-09T16:09:11.966-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2994)
[2024-09-09T16:09:11.967-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2983)
[2024-09-09T16:09:11.968-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
[2024-09-09T16:09:11.970-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:989)
[2024-09-09T16:09:11.972-0300] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2393)
[2024-09-09T16:09:11.974-0300] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2414)
[2024-09-09T16:09:11.976-0300] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2433)
[2024-09-09T16:09:11.978-0300] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2458)
[2024-09-09T16:09:11.978-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$1(RDD.scala:1039)
[2024-09-09T16:09:11.979-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
[2024-09-09T16:09:11.980-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
[2024-09-09T16:09:11.981-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.withScope(RDD.scala:410)
[2024-09-09T16:09:11.982-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.foreachPartition(RDD.scala:1037)
[2024-09-09T16:09:11.983-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.$anonfun$foreachPartition$1(Dataset.scala:3516)
[2024-09-09T16:09:11.984-0300] {subprocess.py:93} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2024-09-09T16:09:11.985-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.$anonfun$withNewRDDExecutionId$1(Dataset.scala:4311)
[2024-09-09T16:09:11.986-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:125)
[2024-09-09T16:09:11.987-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:201)
[2024-09-09T16:09:11.988-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:108)
[2024-09-09T16:09:11.989-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:900)
[2024-09-09T16:09:11.990-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:66)
[2024-09-09T16:09:11.991-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.withNewRDDExecutionId(Dataset.scala:4309)
[2024-09-09T16:09:11.992-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.foreachPartition(Dataset.scala:3516)
[2024-09-09T16:09:11.993-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.saveTable(JdbcUtils.scala:903)
[2024-09-09T16:09:11.994-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcRelationProvider.createRelation(JdbcRelationProvider.scala:65)
[2024-09-09T16:09:11.995-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.SaveIntoDataSourceCommand.run(SaveIntoDataSourceCommand.scala:48)
[2024-09-09T16:09:11.996-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.command.ExecutedCommandExec.sideEffectResult$lzycompute(commands.scala:75)
[2024-09-09T16:09:11.997-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.command.ExecutedCommandExec.sideEffectResult(commands.scala:73)
[2024-09-09T16:09:11.998-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.command.ExecutedCommandExec.executeCollect(commands.scala:84)
[2024-09-09T16:09:11.999-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.QueryExecution$$anonfun$eagerlyExecuteCommands$1.$anonfun$applyOrElse$1(QueryExecution.scala:107)
[2024-09-09T16:09:12.000-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:125)
[2024-09-09T16:09:12.001-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:201)
[2024-09-09T16:09:12.002-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:108)
[2024-09-09T16:09:12.003-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:900)
[2024-09-09T16:09:12.004-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:66)
[2024-09-09T16:09:12.005-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.QueryExecution$$anonfun$eagerlyExecuteCommands$1.applyOrElse(QueryExecution.scala:107)
[2024-09-09T16:09:12.006-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.QueryExecution$$anonfun$eagerlyExecuteCommands$1.applyOrElse(QueryExecution.scala:98)
[2024-09-09T16:09:12.007-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformDownWithPruning$1(TreeNode.scala:461)
[2024-09-09T16:09:12.008-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(origin.scala:76)
[2024-09-09T16:09:12.009-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.trees.TreeNode.transformDownWithPruning(TreeNode.scala:461)
[2024-09-09T16:09:12.010-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.org$apache$spark$sql$catalyst$plans$logical$AnalysisHelper$$super$transformDownWithPruning(LogicalPlan.scala:32)
[2024-09-09T16:09:12.011-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDownWithPruning(AnalysisHelper.scala:267)
[2024-09-09T16:09:12.012-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDownWithPruning$(AnalysisHelper.scala:263)
[2024-09-09T16:09:12.013-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDownWithPruning(LogicalPlan.scala:32)
[2024-09-09T16:09:12.014-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDownWithPruning(LogicalPlan.scala:32)
[2024-09-09T16:09:12.015-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.trees.TreeNode.transformDown(TreeNode.scala:437)
[2024-09-09T16:09:12.016-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.QueryExecution.eagerlyExecuteCommands(QueryExecution.scala:98)
[2024-09-09T16:09:12.017-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.QueryExecution.commandExecuted$lzycompute(QueryExecution.scala:85)
[2024-09-09T16:09:12.018-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.QueryExecution.commandExecuted(QueryExecution.scala:83)
[2024-09-09T16:09:12.019-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.QueryExecution.assertCommandExecuted(QueryExecution.scala:142)
[2024-09-09T16:09:12.020-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameWriter.runCommand(DataFrameWriter.scala:859)
[2024-09-09T16:09:12.021-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameWriter.saveToV1Source(DataFrameWriter.scala:388)
[2024-09-09T16:09:12.022-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameWriter.saveInternal(DataFrameWriter.scala:361)
[2024-09-09T16:09:12.022-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameWriter.save(DataFrameWriter.scala:248)
[2024-09-09T16:09:12.023-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameWriter.jdbc(DataFrameWriter.scala:756)
[2024-09-09T16:09:12.024-0300] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2024-09-09T16:09:12.025-0300] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2024-09-09T16:09:12.026-0300] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2024-09-09T16:09:12.027-0300] {subprocess.py:93} INFO - 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[2024-09-09T16:09:12.028-0300] {subprocess.py:93} INFO - 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2024-09-09T16:09:12.029-0300] {subprocess.py:93} INFO - 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-09-09T16:09:12.030-0300] {subprocess.py:93} INFO - 	at py4j.Gateway.invoke(Gateway.java:282)
[2024-09-09T16:09:12.031-0300] {subprocess.py:93} INFO - 	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2024-09-09T16:09:12.032-0300] {subprocess.py:93} INFO - 	at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2024-09-09T16:09:12.033-0300] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-09-09T16:09:12.034-0300] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-09-09T16:09:12.035-0300] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2024-09-09T16:09:12.036-0300] {subprocess.py:93} INFO - Caused by: org.apache.spark.SparkException: Parquet column cannot be converted in file file:///mnt/c/users/teu20/Documents/Poli/Dados/Raw/empregados/part-00000-ef92445d-d373-4a30-bbf7-dd1f523fde46-c000.snappy.parquet. Column: [reviews_count], Expected: int, Found: BINARY.
[2024-09-09T16:09:12.037-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.errors.QueryExecutionErrors$.unsupportedSchemaColumnConvertError(QueryExecutionErrors.scala:855)
[2024-09-09T16:09:12.038-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:287)
[2024-09-09T16:09:12.039-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-09-09T16:09:12.040-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.FileSourceScanExec$$anon$1.hasNext(DataSourceScanExec.scala:593)
[2024-09-09T16:09:12.041-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.columnartorow_nextBatch_0$(Unknown Source)
[2024-09-09T16:09:12.042-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-09-09T16:09:12.043-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-09-09T16:09:12.044-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-09-09T16:09:12.045-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-09-09T16:09:12.046-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-09-09T16:09:12.047-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator.isEmpty(Iterator.scala:387)
[2024-09-09T16:09:12.048-0300] {subprocess.py:93} INFO - 	at scala.collection.Iterator.isEmpty$(Iterator.scala:387)
[2024-09-09T16:09:12.049-0300] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.isEmpty(Iterator.scala:1431)
[2024-09-09T16:09:12.050-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.savePartition(JdbcUtils.scala:684)
[2024-09-09T16:09:12.051-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.$anonfun$saveTable$1(JdbcUtils.scala:904)
[2024-09-09T16:09:12.052-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.$anonfun$saveTable$1$adapted(JdbcUtils.scala:903)
[2024-09-09T16:09:12.053-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2(RDD.scala:1039)
[2024-09-09T16:09:12.054-0300] {subprocess.py:93} INFO - 	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2$adapted(RDD.scala:1039)
[2024-09-09T16:09:12.055-0300] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.$anonfun$runJob$5(SparkContext.scala:2433)
[2024-09-09T16:09:12.056-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-09-09T16:09:12.057-0300] {subprocess.py:93} INFO - 	at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-09-09T16:09:12.058-0300] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-09-09T16:09:12.059-0300] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-09-09T16:09:12.060-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-09-09T16:09:12.061-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-09-09T16:09:12.062-0300] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-09-09T16:09:12.063-0300] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-09-09T16:09:12.064-0300] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2024-09-09T16:09:12.065-0300] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2024-09-09T16:09:12.066-0300] {subprocess.py:93} INFO - 	... 1 more
[2024-09-09T16:09:12.067-0300] {subprocess.py:93} INFO - Caused by: org.apache.spark.sql.execution.datasources.SchemaColumnConvertNotSupportedException: column: [reviews_count], physicalType: BINARY, logicalType: int
[2024-09-09T16:09:12.068-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.ParquetVectorUpdaterFactory.constructConvertNotSupportedException(ParquetVectorUpdaterFactory.java:1136)
[2024-09-09T16:09:12.069-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.ParquetVectorUpdaterFactory.getUpdater(ParquetVectorUpdaterFactory.java:199)
[2024-09-09T16:09:12.070-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedColumnReader.readBatch(VectorizedColumnReader.java:175)
[2024-09-09T16:09:12.071-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader.nextBatch(VectorizedParquetRecordReader.java:342)
[2024-09-09T16:09:12.072-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader.nextKeyValue(VectorizedParquetRecordReader.java:233)
[2024-09-09T16:09:12.073-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.RecordReaderIterator.hasNext(RecordReaderIterator.scala:39)
[2024-09-09T16:09:12.074-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-09-09T16:09:12.075-0300] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:283)
[2024-09-09T16:09:12.076-0300] {subprocess.py:93} INFO - 	... 28 more
[2024-09-09T16:09:12.077-0300] {subprocess.py:93} INFO - 
[2024-09-09T16:09:12.338-0300] {subprocess.py:97} INFO - Command exited with return code 1
[2024-09-09T16:09:12.347-0300] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/operators/bash.py", line 249, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2024-09-09T16:09:12.364-0300] {logging_mixin.py:190} INFO - Task instance in failure state
[2024-09-09T16:09:12.365-0300] {logging_mixin.py:190} INFO - Task start:2024-09-09 19:08:58.670930+00:00 end:2024-09-09 19:09:12.364166+00:00 duration:13.693236
[2024-09-09T16:09:12.366-0300] {logging_mixin.py:190} INFO - Task:<Task(BashOperator): dag_save_database_raw> dag:<DAG: dag_save_database_raw> dagrun:<DagRun dag_save_database_raw @ 2024-08-21 08:15:00+00:00: scheduled__2024-08-21T08:15:00+00:00, state:running, queued_at: 2024-09-09 19:08:44.648755+00:00. externally triggered: False>
[2024-09-09T16:09:12.367-0300] {logging_mixin.py:190} INFO - Failure caused by Bash command failed. The command returned a non-zero exit code 1.
[2024-09-09T16:09:12.368-0300] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=dag_save_database_raw, task_id=dag_save_database_raw, run_id=scheduled__2024-08-21T08:15:00+00:00, execution_date=20240821T081500, start_date=20240909T190858, end_date=20240909T190912
[2024-09-09T16:09:12.393-0300] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-09-09T16:09:12.395-0300] {standard_task_runner.py:124} ERROR - Failed to execute job 31 for task dag_save_database_raw (Bash command failed. The command returned a non-zero exit code 1.; 132727)
Traceback (most recent call last):
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
  File "/mnt/c/Users/teu20/Documents/Poli/Repo/eEDB011-20243/atividade 6 - WSL/airflow/airflow_env/lib/python3.10/site-packages/airflow/operators/bash.py", line 249, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2024-09-09T16:09:12.443-0300] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-09-09T16:09:12.455-0300] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-09-09T16:09:12.457-0300] {local_task_job_runner.py:245} INFO - ::endgroup::
